### PR TITLE
Add documentation about "oc image mirror" commmand

### PR DESCRIPTION
**Pull request by legionus**
@bparees Please review.

### CODE REVIEW

File: install_config/registry/accessing_registry.adoc [hunk 1]
```diff
@@ -344,3 +344,50 @@ $ oc adm policy add-cluster-role-to-user prometheus-scraper <username>
 ----
 
 See the link:https://prometheus.io/docs/querying/basics/[upstream Prometheus documentation] for more advanced queries and recommended visualizers.
+
+[[mirror-registry-images]]
```
Comment (bparees) [inline install_config/registry/accessing_registry.adoc]: i think i would have expected to find this under https://docs.openshift.org/latest/dev_guide/managing_images.html

(https://docs.openshift.org/latest/install_config/registry/accessing_registry.html#access-pushing-and-pulling-images should probably be moved there too, but that is part of a larger refactor that i think needs to happen w/ this content)

File: install_config/registry/accessing_registry.adoc [hunk 2]
```diff
@@ -344,3 +344,50 @@ $ oc adm policy add-cluster-role-to-user prometheus-scraper <username>
...
+----
+
+[NONE]
+====
+The `oc image mirror` runs locally, not inside the openshift cluster. This means
+that this command must have access to the source and destination registry.
+====
+
+If the Docker registry requires authentication to pull or push an image, you
+need to log in manually before executing the command.
```
Comment (bparees) [inline install_config/registry/accessing_registry.adoc]: "you need to log in manually (using the `docker login` command) before executing the `oc image mirror` command."

File: install_config/registry/accessing_registry.adoc [hunk 3]
```diff
@@ -344,3 +344,50 @@ $ oc adm policy add-cluster-role-to-user prometheus-scraper <username>
...
+
+[[mirror-registry-images-s3]]
+=== Mirroring Images to S3
+
+It is possible to copy the image directly to the S3 bucket. In this case the S3
```
Comment (bparees) [inline install_config/registry/accessing_registry.adoc]: s/to the S3 bucket/to an S3 bucket/

File: install_config/registry/accessing_registry.adoc [hunk 4]
```diff
@@ -344,3 +344,50 @@ $ oc adm policy add-cluster-role-to-user prometheus-scraper <username>
...
+region and bucket must be the first two segments after the host.
+
+[source, bash]
+----
+$ oc image mirror 172.30.124.220:5000/myproject/busybox:latest s3://s3.amazonaws.com/us-east-1/mybucket/image:latest
+----
+
+After that, you can pull the image from S3 bucket without using any registry
```
Comment (bparees) [inline install_config/registry/accessing_registry.adoc]: s/from S3/from the S3/

Comment (bparees) [reply install_config/registry/accessing_registry.adoc]: can you provide an example of the docker pull invocation you'd use to do that?

**Review by bparees — CHANGES_REQUESTED** [review]

File: install_config/registry/accessing_registry.adoc (hunk 3 above)
Comment (bparees) [reply install_config/registry/accessing_registry.adoc]: where do the s3 credentials come from when doing this?

File: install_config/registry/accessing_registry.adoc (hunk 1 above)
Comment (legionus) [reply install_config/registry/accessing_registry.adoc]: @bparees This has nothing to do with openshift images. This command is our clone of skopeo. You do not need to have imagestream to use this command. This is just client side general tool.

Comment (bparees) [reply install_config/registry/accessing_registry.adoc]: yes, but it also has nothing to do w/ the openshift registry (you can use it to mirror content between any two registries).

At least the images section is about managing images (not necessarily imagestreams).  Plus it's something a developer, not an administrator, would be doing.

Comment (legionus) [reply install_config/registry/accessing_registry.adoc]: @bparees OK. I will move this to managing_images.

**Comment by legionus** [comment]
@bparees Updated

File: install_config/registry/accessing_registry.adoc (hunk 4 above)
Comment (legionus) [reply install_config/registry/accessing_registry.adoc]: @bparees I can't make it work with `minio` and `s3.amazonaws.com`. So, I can't give any examples here.

Comment (bparees) [reply install_config/registry/accessing_registry.adoc]: what is preventing you from making it work w/ s3?

File: dev_guide/managing_images.adoc [hunk 5]
```diff
@@ -1105,6 +1105,86 @@ my-image   172.30.48.125:5000/test/my-image   latest    3 seconds ago
...
+
+[source, bash]
+----
+$ oc image mirror docker.io/library/busybox:latest s3://s3.amazonaws.com/us-east-1/mybucket/localhost.localdomain/toybox:latest
+----
+
+[NOTE]
+----
+After bucket component you should specify `localhost` or any valid domain name
+with dot (`.`).
```
Comment (bparees) [inline dev_guide/managing_images.adoc]: "with a dot"

File: dev_guide/managing_images.adoc [hunk 6]
```diff
@@ -1105,6 +1105,86 @@ my-image   172.30.48.125:5000/test/my-image   latest    3 seconds ago
...
+----
+
+After that, you can pull the image from the S3 bucket without using any registry
```
Comment (bparees) [inline dev_guide/managing_images.adoc]: s/any/a/

File: dev_guide/managing_images.adoc (hunk 5 above)
Comment (legionus) [reply dev_guide/managing_images.adoc]: @bparees Fixed

File: dev_guide/managing_images.adoc (hunk 6 above)
Comment (legionus) [reply dev_guide/managing_images.adoc]: @bparees Fixed

File: dev_guide/managing_images.adoc [hunk 7]
```diff
@@ -1105,6 +1105,85 @@ my-image   172.30.48.125:5000/test/my-image   latest    3 seconds ago
...
+The `oc image mirror` runs locally, not inside the openshift cluster. This means
+that this command must have access to the source and destination registry.
+====
+
+If the Docker registry requires authentication to pull or push an image, you
+you need to log in manually (using the `docker login` command) before executing
+the oc image mirror command.
+
+[[mirror-registry-images-s3]]
+=== Mirroring Images to S3
```
Comment (bparees) [inline dev_guide/managing_images.adoc]: per discussion w/ @smarterclayton about this and the issues you ran into with it, let's remove the s3-related bits of this doc.

Comment (legionus) [reply dev_guide/managing_images.adoc]: @bparees done

**Comment by bparees** [comment]
lgtm

@openshift/team-documentation ptal

**Comment by bmcelvee** [comment]
@legionus the content looks good. Once I'm able to merge this PR (when we have a 3.11 branch), I'll open a follow-up PR with some minor edits. 

In the meantime, @dongboyan77 PTAL. 

Thanks!

**Comment by bmcelvee** [comment]
@dongboyan77 PTAL

**Comment by bmcelvee** [comment]
@dongboyan77 PTAL :sun_with_face:

**Comment by dongboyan77** [comment]
LGTM, thanks

**Pull request merged**